File: 5.prefix_sums/lesson_5_min_avg_two_slice.py
def solution(A):
    s = len(A)

    if(s == 2):
        return 0

    prefixes = [0]*s
    prefixes[0] = A[0]

    for index in range(1, s):
        prefixes[index] = prefixes[index-1] + A[index]

    min_value = 11000
    min_index = -1

    for i in range(s-1):
        maxi = i+3
        if(maxi > s):
            maxi = s
        for j in range(i+1, maxi):
            summ = prefixes[j] 
            if(i > 0):
                summ -= prefixes[i-1]
            current_avg = summ / (j - i + 1)
            if(current_avg < min_value):
                min_value = current_avg
                min_index = i
    return min_index

File: 5.prefix_sums/lesson_5_min_avg_two_slice.js
function solution(A) {
    let n = A.length
    let min = 10000 * n
    let minIndex = 0

    for(let i = 0; i < n - 1; i++) {
        let sum = A[i]
        let end = i + 4 > n ? n : i + 4
        for(let j = i + 1; j < end; j++) {
            sum += A[j]
            let div = sum / (j - i + 1)
            if(div < min) {
                min = div
                minIndex = i
            }
        }
    }
    return minIndex
}